### PR TITLE
App models

### PR DESCRIPTION
**Pull request by daryltanwk**

### CODE REVIEW

File: frontend/.angular-cli.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
       "prefix": "app",
       "styles": [
         "styles.css",
-        "../node_modules/bootswatch/readable/bootstrap.min.css"
+        "../node_modules/bootswatch/readable/bootstrap.min.css",
```
Comment (soedar) [inline frontend/.angular-cli.json]: Just checking, but is this the right way to reference a css file from a dependency?

File: frontend/src/app/play/play.component.ts [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+  mockId = 'holder';
+
+  constructor(private playersService: PlayersService, private lobbyService: LobbyService) { }
 
   ngOnInit() {
+    this.lobbies = this.lobbyService.getLobbies();
+  }
+
+  lobbySelected() {
+    if (!this.selectedLobby) {
```
Comment (soedar) [inline frontend/src/app/play/play.component.ts]: Probably can condense the if/else to `return this.selectedLobby;` ?

File: frontend/src/app/play/play.component.ts [hunk 3]
```diff
@@ -1,17 +1,53 @@
...
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  shouldHighlight(id: string) {
+    if (typeof this.selectedLobby !== 'undefined' && id === this.selectedLobby.id) {
```
Comment (soedar) [inline frontend/src/app/play/play.component.ts]: Also can probably condense this if/else to just the condition inside the `if`

Comment (soedar) [inline frontend/src/app/play/play.component.ts]: also, is there a difference between checking the typeof and checking `this.selectedLobby !== undefined` ?

File: frontend/src/app/services/lobby.service.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  createLobby(name: string) {
+    // insert actual backend call here
+    this.lobbies.push(new Lobby(this.lobbies.length.toString(), name));
+  }
+
+  removeLobby(id) {
+    // insert actual backend call here
+    const index = this.lobbies.findIndex((lobby) => {
+      if (lobby.lobbyId() === id) {
```
Comment (soedar) [inline frontend/src/app/services/lobby.service.ts]: can condense this if/else also

File: frontend/src/app/services/players.service.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  addPlayer(playerName: string) {
+    // insert actual backend call here
+    const player = new Player(this.players.length.toString(), playerName);
+    this.players.push(player);
+  }
+
+  removePlayer(playerId: string) {
+    // insert actual backend call here
+    const index = this.players.findIndex((player) => {
```
Comment (soedar) [inline frontend/src/app/services/players.service.ts]: can condense this

File: frontend/src/app/services/lobby.service.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        return false;
+      }
+    });
+
+    if (typeof index !== 'undefined') {
+      this.lobbies.splice(index, 1);
+    }
+  }
+
+  playerJoins(playerId: string, lobbyId: string) {
```
Comment (soedar) [inline frontend/src/app/services/lobby.service.ts]: Not really sure, but should the logic of player joining lobby be here or inside the model? How does the service work?

File: frontend/.angular-cli.json (hunk 1 above)
Comment (daryltanwk) [inline frontend/.angular-cli.json]: i think so. that's how the guy in the video did it. and i guess by using it this way, the css kinda gets updated if u ran npm update...

File: frontend/src/app/play/play.component.ts (hunk 2 above)
Comment (daryltanwk) [inline frontend/src/app/play/play.component.ts]: how does that get me a boolean?

File: frontend/src/app/play/play.component.ts (hunk 3 above)
Comment (daryltanwk) [inline frontend/src/app/play/play.component.ts]: sadly, doing `this.selectedLobby` when it is undefined throws error that im referring to something that isnt defined in chrome console. but the `typeof this.selectedLobby` gives me a string. so no errors generated.

Comment (soedar) [inline frontend/src/app/play/play.component.ts]: I see, ok

**Pull request merged**